public/app/features/variables/editor/reducer.ts

```diff
@@ -10,7 +13,12 @@ export interface VariableEditorState<ExtendedProps extends {} = {}> {
   extended: VariableEditorExtension<ExtendedProps> | null;
 }
 
-export const initialVariableEditorState: VariableEditorState = {
+export interface QueryVariableEditorState {
+  dataSource?: DataSourceApi | null;
+  VariableQueryEditor?: VariableQueryEditorType;
+}
+
+export const initialVariableEditorState: VariableEditorState<QueryVariableEditorState> = {
```

Comment:
The typing here is a little bit weird, and I'm not sure if I full understand what interfaces are reused elsewhere.

Is there something I'm missing that prevents us from dropping the VariableEditorExtension type and directly typing the extended properties?

Reply:
The extended properties change depending on which variable type that is in the editor.